concent_api/api_testing_common.py

```diff
@@ -165,6 +165,8 @@ def _print_data(data, url):
     print()
     if response.status_code in [202, 204]:
         return None
+    elif response.status_code == 400:
+        return json.loads(response._content)
```

Comment:
You should use `Content-Type` header to decide how to decode the response, not the error code. Responses with the same code can have different content in different situations.

Reply:
BTW, you can do it as a separate pull request. I'm sure we have this problem in many more places and it would be a good idea to fix them all.

Reply:
resolved in https://github.com/golemfactory/concent/pull/773